### PR TITLE
[stdlib] Implement customization points for StrideTo, StrideThrough, and LazyFilterX

### PR DESCRIPTION
**Pull request by moiseev**

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please clean test Linux platform

**Comment by moiseev** [comment]
@swift-ci Please smoke benchmark

**Comment by moiseev** [comment]
@swift-ci Please smoke benchmark

**Comment by moiseev** [comment]
@swift-ci Please test

File: stdlib/public/core/Stride.swift.gyb [hunk 1]
```diff
@@ -234,6 +234,23 @@ public struct StrideTo<Element : Strideable> : Sequence, CustomReflectable {
...
+    _ preprocess: () throws -> R
+  ) rethrows -> R? {
+    return try preprocess()
+  }
+
+  @_inlineable
+  public func _customContainsEquatableElement(
+    _ element: Element
+  ) -> Bool? {
+    if element < _start || _end <= element {
```
Comment (airspeedswift) [inline stdlib/public/core/Stride.swift.gyb]: Is it worth adding an overload for when `Element: FixedWidthInteger` that would compute contains entirely?

**Review by airspeedswift — APPROVED** [review]

File: stdlib/public/core/Stride.swift.gyb (hunk 1 above)
Comment (moiseev) [reply stdlib/public/core/Stride.swift.gyb]: Hm...

**Comment by airspeedswift** [comment]
@swift-ci please smoke test compiler performance

**Pull request merged**